File: mobile/lib/models/university.dart
/// Universitas Indonesia dari endpoint publik `/universities`.
class University {
  final String id;
  final String code;
  final String name;
  final String? type;
  final String? province;
  final String? city;

  const University({
    required this.id,
    required this.code,
    required this.name,
    this.type,
    this.province,
    this.city,
  });

  factory University.fromJson(Map<String, dynamic> json) {
    return University(
      id: json['id'] as String? ?? '',
      code: json['code'] as String? ?? '',
      name: json['name'] as String? ?? '',
      type: json['type'] as String?,
      province: json['province'] as String?,
      city: json['city'] as String?,
    );
  }
}

/// Program studi (prodi) dari endpoint `/universities/{id}/study-programs`.
class StudyProgram {
  final String id;
  final String name;

  const StudyProgram({required this.id, required this.name});

  factory StudyProgram.fromJson(Map<String, dynamic> json) {
    return StudyProgram(
      id: json['id'] as String? ?? '',
      name: json['name'] as String? ?? '',
    );
  }
}

File: backend/app/Models/University.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Attributes\Fillable;
use Illuminate\Database\Eloquent\Concerns\HasUuids;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;

#[Fillable(['code', 'name', 'type', 'province', 'city'])]
class University extends Model
{
    use HasUuids;

    /**
     * Study programs (prodi) offered by this university.
     */
    public function studyPrograms(): HasMany
    {
        return $this->hasMany(StudyProgram::class);
    }
}
